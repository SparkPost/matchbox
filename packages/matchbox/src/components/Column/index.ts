export { default as Column } from './Column';
export * from './Column';
