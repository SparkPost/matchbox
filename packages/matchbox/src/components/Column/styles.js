import { system } from 'styled-system';

export const gutter = system({
  gutter: {
    property: 'marginRight',
    scale: 'space',
  },
});
