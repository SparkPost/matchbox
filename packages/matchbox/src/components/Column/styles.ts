import { system } from 'styled-system';

export const gutter = system({
  gutter: {
    property: 'marginLeft',
    scale: 'space',
  },
});

export const columnWidth = system({
  width: {
    property: 'width',
    scale: 'layout',
  },
});
