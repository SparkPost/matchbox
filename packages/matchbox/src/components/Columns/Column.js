import React from 'react';

function Column(props) {
  const { children } = props;

  return <div>{children}</div>;
}

Column.displayName = 'Columns.Column';

export default Column;
