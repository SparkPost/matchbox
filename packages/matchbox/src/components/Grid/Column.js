import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './Grid.module.scss';

class Column extends Component {
  static displayName = 'Grid.Column';

  static propTypes = {
    xs: PropTypes.number,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
    xsOffset: PropTypes.number,
    smOffset: PropTypes.number,
    mdOffset: PropTypes.number,
    lgOffset: PropTypes.number,
    children: PropTypes.node
  };

  render() {
    const {
      children,
      xs, sm, md, lg, xl,
      xsOffset, smOffset, mdOffset, lgOffset, xlOffset,
      ...rest
    } = this.props;

    const colClasses = classnames(
      styles.Column,
      xs && styles[`xs-${xs}`],
      sm && styles[`sm-${sm}`],
      md && styles[`md-${md}`],
      lg && styles[`lg-${lg}`],
      xl && styles[`xl-${xl}`],
      xsOffset && styles[`xs-offset-${xsOffset}`],
      smOffset && styles[`sm-offset-${smOffset}`],
      mdOffset && styles[`md-offset-${mdOffset}`],
      lgOffset && styles[`lg-offset-${lgOffset}`],
      xlOffset && styles[`xl-offset-${xlOffset}`]
    );

    return <div className={colClasses} {...rest}>{children}</div>;
  }
}

export default Column;
