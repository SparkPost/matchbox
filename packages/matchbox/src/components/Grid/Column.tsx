import React from 'react';
import styled from 'styled-components';
import { gridColumnStyle } from './styles';

const StyledGridColumn = styled('div')`
  ${gridColumnStyle}
`;

type GridColumnProps = {
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  xsOffset?: number;
  smOffset?: number;
  mdOffset?: number;
  lgOffset?: number;
  children?: React.ReactNode;
};

function Column({ children, ...props }: GridColumnProps): JSX.Element {
  return <StyledGridColumn {...props}>{children}</StyledGridColumn>;
}

Column.displayName = 'Grid.Column';
export default Column;
