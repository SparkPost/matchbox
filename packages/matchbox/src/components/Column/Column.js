import React from 'react';
import PropTypes from 'prop-types';
import { ColumnsContext } from '../Columns/context';
import styled from 'styled-components';
import { Box } from '../Box';
import { gutter } from './styles';

const StyledColumn = styled(Box)`
  ${gutter}
`;

const Column = React.forwardRef(function Column(props, ref) {
  const { width, children } = props;
  const { space, collapsed } = React.useContext(ColumnsContext);

  let columnWidth = width;

  if (collapsed) {
    columnWidth = '100%';
  }

  if (width === 'content') {
    columnWidth = 'auto';
  }

  return (
    <StyledColumn
      width={columnWidth}
      flex={!width && !collapsed ? '1' : ''}
      gutter={space}
      ref={ref}
    >
      {children}
    </StyledColumn>
  );
});

Column.displayName = 'Column';

Column.propTypes = {
  children: PropTypes.node,
  width: PropTypes.oneOfType([PropTypes.oneOf(['content']), PropTypes.number]),
};

export default Column;
