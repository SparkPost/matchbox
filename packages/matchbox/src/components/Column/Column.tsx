import React from 'react';
import { display, DisplayProps } from 'styled-system';
import { ColumnsContext } from '../Columns/context';
import styled from 'styled-components';
import { ResponsiveValue } from 'styled-system';
import { SpaceKeys } from '../ThemeProvider/theme';
import { Box } from '../Box';
import { gutter } from './styles';

const StyledColumn = styled(Box)`
  ${display}
  ${gutter}

  &:first-child {
    padding-top: 0;
  }
`;

type ColumnProps = {
  children?: React.ReactNode;
  width?: 'content' | ResponsiveValue<SpaceKeys | number>;
  'data-id'?: string;
  className?: string;
  reverse?: ResponsiveValue<boolean>;
} & DisplayProps;

const Column = React.forwardRef(function Column(props: ColumnProps, ref) {
  const { width, children, className, display } = props;
  const { space, collapsed } = React.useContext(ColumnsContext);

  let widthOverride = null;

  if (collapsed) {
    widthOverride = '100%';
  }

  if (width === 'content') {
    widthOverride = 'auto';
  }

  return (
    <StyledColumn
      display={display}
      data-id={props['data-id']}
      className={className}
      width={widthOverride || width}
      flex={!width && !collapsed ? '1' : ''}
      pt={collapsed ? space : null}
      gutter={space}
      ref={ref}
      tabIndex="-1"
    >
      {children}
    </StyledColumn>
  );
}) as React.ForwardRefExoticComponent<ColumnProps>;

Column.displayName = 'Column';

export default Column;
