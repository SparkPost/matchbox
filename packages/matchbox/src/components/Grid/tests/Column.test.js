import React from 'react';
import Column from '../Column';
import { shallow } from 'enzyme';

describe('Column', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Column><p>Grid</p></Column>);
  });

  it('renders with default props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with sizing prop', () => {
    wrapper.setProps({ xs: 12, sm: 6, md: 6, lg: 4, xl: 2, xsOffset: 1, smOffset: 2, mdOffset: 2, lgOffset: 3, xlOffset: 4 });
    expect(wrapper).toMatchSnapshot();
  });
});
