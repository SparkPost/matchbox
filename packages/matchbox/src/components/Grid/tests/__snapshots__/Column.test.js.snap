// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders with default props 1`] = `
<div
  className="Column"
>
  <p>
    Grid
  </p>
</div>
`;

exports[`Column renders with sizing prop 1`] = `
<div
  className="Column xs-12 sm-6 md-6 lg-4 xl-2 xs-offset-1 sm-offset-2 md-offset-2 lg-offset-3 xl-offset-4"
>
  <p>
    Grid
  </p>
</div>
`;
